v2/types.go

```diff
@@ -995,3 +995,59 @@ func ConvertBlockItems(input []*pb.BlockItem) []*BlockItem {
 
 	return result
 }
+
+// A chain epoch.
+type Epoch struct {
+	Value uint64
+}
+
+// Input to queries which take an epoch as a parameter.
+type isEpochRequest interface {
+	isEpochRequest()
+}
+
+// Query for the epoch of a specified block.
+type EpochRequestBlockHash struct {
+	// The block to query at.
+	BlockHash isBlockHashInput
+}
+
+func (EpochRequestBlockHash) isEpochRequest() {}
```

Comment:
I don't understand what this function  is supposed to do.

Reply:
It implements the `isEpochRequest` interface. See above comment for explanation of why it is done like this.